apm-sniffer/apm-sdk-plugin/pulsar-plugin/src/main/java/org/apache/skywalking/apm/plugin/pulsar/MessageConstructorInterceptor.java

```diff
@@ -34,6 +35,12 @@
 
     @Override
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
-        objInst.setSkyWalkingDynamicField(new MessageEnhanceRequiredInfo());
+        final Object msgArgument = allArguments[2];
+        if (objInst instanceof TopicMessageImpl && msgArgument instanceof EnhancedInstance) {
```

Comment:
If you need to separate the type of current object, why not use two different instrumentation definitions?

Reply:
The enhanced class is an interface, I want to enhance all the implementation classes, but some of them are used for wrapping other implementation, so they are only distinguished by the parameters of the constructor instead of using another instrumentation definitions.

Reply:
I was talking about objInst. 
Are you saying you could give a full list of implementors? Interface based is not a stable, you don't know how people will or have extended.

Reply:
Yes, it is indeed not stable. The specific types are also checked in the pulsar client code, and I think I can change the enhanced class into two certain types of implementors.

Reply:
Thanks. Targeting the specific classes is already a better idea, and the codes would be more clear and have better performance.

Reply:
done

Reply:
Thanks. Let's wait for @codelipenghui 's review.